src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -3202,12 +3202,9 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
     def typedEtaExpansion(tree: Tree, mode: Mode, pt: Type): Tree = {
       debuglog(s"eta-expanding $tree: ${tree.tpe} to $pt")
 
-      if (tree.tpe.isDependentMethodType) DependentMethodTpeConversionToFunctionError(tree, tree.tpe) // TODO: support this
-      else {
-        val expansion = etaExpand(tree, context.owner)
-        if (context.undetparams.isEmpty) typed(expansion, mode, pt)
-        else instantiate(typed(expansion, mode), mode, pt)
-      }
+      val expansion = etaExpand(tree, context.owner)
+      if (context.undetparams.isEmpty) typed(expansion, mode, pt)
+      else instantiate(typed(expansion, mode), mode, pt)
```

Comment:
I don't understand the restriction yet, but `etaExpand` still has adriaanm comment that not supported.

Reply:
Thanks! I'll be off for a week 🇷🇴 and don't have more time right now to try to break it, will get back to it after.

Reply:
Enjoy! Insert joke about children as dependent types.

Reply:
Added a few more tests, AFAICT `etaExpand` works well with dependent method types